src/xenia/base/memory.cc

```diff
@@ -215,7 +219,102 @@ void copy_and_swap_16_in_32_unaligned(void* dest_ptr, const void* src_ptr,
     dest[i] = (src[i] >> 16) | (src[i] << 16);
   }
 }
+
+#elif XE_ARCH_ARM64
+
+// Although NEON offers vector rev instructions (like vrev32q_u8), they are
+// slower in benchmarks. Also, using uint8x16xN_t wasn't any faster in the
+// benchmarks, hence we use just use one SIMD register to minimize residual
+// processing.
+
+void copy_and_swap_16_aligned(void* dst_ptr, const void* src_ptr,
+                              size_t count) {
+  copy_and_swap_16_unaligned(dst_ptr, src_ptr, count);
+}
+
+void copy_and_swap_16_unaligned(void* dst_ptr, const void* src_ptr,
+                                size_t count) {
+  auto dst = reinterpret_cast<uint16_t*>(dst_ptr);
+  auto src = reinterpret_cast<const uint16_t*>(src_ptr);
+
+  const uint8x16_t tbl_idx = vcombine_u8(vcreate_u8(0x0607040502030001),
```

Comment:
For safety across compilers — though maybe literals of this width are fine everywhere, but to explicitly disambiguate anyway — wrapping these literals in `UINT64_C` (from `<cstdint>`) may be desirable.